src/utp_stream.cpp

```diff
@@ -89,6 +88,29 @@ void utp_log(char const* fmt, ...)
 	va_end(l);
 }
 
+bool is_utp_stream_logging() {
+	return log_file_holder.utp_log_file != NULL;
+}
+
+void set_utp_stream_logging(bool enable) {
+	if (enable)
+	{
+		if (log_file_holder.utp_log_file == NULL)
+		{
+			log_file_holder.utp_log_file = fopen("utp.log", "w+");
+		}
+	}
+	else
+	{
+		if (log_file_holder.utp_log_file != NULL)
+		{
+			FILE* f = log_file_holder.utp_log_file;
+			log_file_holder.utp_log_file = NULL;
+			fclose(f);
+		}
+	}
+}
+
 #define UTP_LOG utp_log
 #if TORRENT_VERBOSE_UTP_LOG
```

Comment:
I added a new macro TORRENT_UTP_LOG_ENABLE to avoid a lot of source code changes. Let me know.